packages/@aws-cdk/core/lib/asset-staging.ts

```diff
@@ -179,6 +200,7 @@ export class AssetStaging extends Construct {
         workingDirectory: options.workingDirectory ?? AssetStaging.BUNDLING_INPUT_DIR,
       });
     } catch (err) {
+      fs.removeSync(bundleDir);
```

Comment:
This is unrelated to this PR, right? 
I am also not sure we actually want to delete the bundleDir for diagnosability (I would even include it in the error message).

Reply:
It is, see https://github.com/aws/aws-cdk/pull/8916#issuecomment-655004255

> Otherwise, subsequent runs may falsely believe that the `bundleDir` can be re-used.

Rename it in case of failure instead (e.g. append `-error`)?

Reply:
> Rename it in case of failure instead (e.g. append `-error`)?

This seems good to me. But, if we go this route, we might consider removing `-error`  directories if they already exist, such as if docker fails twice for the same bundleDir.

Reply:
For now I've implemented the `-error` suffix. I'm happy to circle back around.